modal/_utils/async_utils.py

```diff
@@ -573,6 +573,34 @@ class StopSentinelType:
 
 
 async def async_merge(*generators: AsyncGenerator[T, None]) -> AsyncGenerator[T, None]:
+    """
+    Asynchronously merges multiple async generators into a single async generator.
+
+    This function takes multiple async generators and yields their values in the order
+    they are produced. If any generator raises an exception, the exception is propagated.
+
+    Args:
+        *generators: One or more async generators to be merged.
+
+    Yields:
+        The values produced by the input async generators.
+
+    Raises:
+        Exception: If any of the input generators raises an exception, it is propagated.
+
+    Example:
+        async def gen1():
```

Comment:
if you add a markdown codefence with ```` ```py````around the example and optionally add assertions at the end, the inlined example will be automatically tested in CI to prevent example-drift if we change something

Reply:
(see https://pypi.org/project/pytest-markdown-docs/)

Reply:
Tried that but looks like all docstring tests are being skipped in CI. Might take a look later.

Reply:
OK hopefully should be fixed in #2576.